pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/BrokersBase.java

```diff
@@ -437,5 +440,62 @@ private void doShutDownBrokerGracefully(int maxConcurrentUnloadPerSec,
         pulsar().getBrokerService().unloadNamespaceBundlesGracefully(maxConcurrentUnloadPerSec, forcedTerminateTopic);
         pulsar().closeAsync();
     }
+
+    /**
+     * dynamically update log4j2 logger level at runtime.
+     *
+     * @param targetClassName : class name to update
+     * @param targetLevel     : target log level
+     */
+    private CompletableFuture<Void> internalUpdateLoggerLevelAsync(String targetClassName, String targetLevel) {
+        CompletableFuture<Void> loggerLevelFuture = new CompletableFuture<>();
+        try {
+            String className;
+            // if set "ROOT" will take effect to rootLogger
+            if (targetClassName.trim().equalsIgnoreCase("ROOT")) {
+                className = LogManager.ROOT_LOGGER_NAME;
+            } else {
+                className = targetClassName;
+            }
+            Level newLevel;
+            try {
+                newLevel = Level.valueOf(targetLevel);
+                Configurator.setAllLevels(className, newLevel);
+            } catch (IllegalArgumentException | NullPointerException e) {
+                // Unknown Log Level or NULL
+                throw new RestException(Status.PRECONDITION_FAILED, "Invalid logger level.");
+            }
+        } catch (RestException re) {
+            throw re;
+        } catch (Exception ie) {
+            throw new RestException(Status.PRECONDITION_FAILED, "Internal error.");
+        }
+        return CompletableFuture.completedFuture(null);
+    }
+
+    @POST
+    @Path("/log4j/{classname}/{level}")
```

Comment:
I wouldn't use 'log4j' in this path.
'logging' is enough.
We are not sure that we will keep log4j forever.
This API is broad enough to be compatible with major logging frameworks in the future